src/pfe/portal/modules/LoadRunner.js

```diff
@@ -289,48 +289,45 @@ module.exports = class LoadRunner {
       await this.waitForHealth(false);
       log.info(`beginJavaProfiling: Starting liberty server`); 
-      await docker.exec(this.project, ['bash', '-c', `source $HOME/artifacts/envvars.sh; $HOME/artifacts/server_setup.sh; cd $WLP_USER_DIR;  /opt/ibm/wlp/bin/server start;`]);
+      const startCommand = ['bash', '-c', `source $HOME/artifacts/envvars.sh; $HOME/artifacts/server_setup.sh; cd $WLP_USER_DIR;  /opt/ibm/wlp/bin/server start; `];
+      await cwUtils.spawnContainerProcess(this.project, startCommand);
       await this.waitForHealth(true);
     }
 
-    const command = `export javapid=(\`pidof java\`); java -jar $JAVA_HOME/jre/lib/ext/healthcenter.jar ID=\${javapid[-1]} level=headless -Dcom.ibm.java.diagnostics.healthcenter.headless.run.number.of.runs=1 -Dcom.ibm.java.diagnostics.healthcenter.headless.run.duration=${durationInMins} -Dcom.ibm.java.diagnostics.healthcenter.headless.output.directory=${loadTestDir} -Dcom.ibm.diagnostics.healthcenter.data.memory=off -Dcom.ibm.diagnostics.healthcenter.data.memorycounters=off -Dcom.ibm.diagnostics.healthcenter.data.cpu=off -Dcom.ibm.diagnostics.healthcenter.data.environment=off -Dcom.ibm.diagnostics.healthcenter.data.locking=off -Dcom.ibm.diagnostics.healthcenter.data.memory=off -Dcom.ibm.diagnostics.healthcenter.data.threads=off`;
-
+    const healthCentercommand = `export javapid=(\`pidof java\`); java -jar $JAVA_HOME/jre/lib/ext/healthcenter.jar ID=\${javapid[-1]} level=headless -Dcom.ibm.java.diagnostics.healthcenter.headless.run.number.of.runs=1 -Dcom.ibm.java.diagnostics.healthcenter.headless.run.duration=${durationInMins} -Dcom.ibm.java.diagnostics.healthcenter.headless.output.directory=${loadTestDir} -Dcom.ibm.diagnostics.healthcenter.data.memory=off -Dcom.ibm.diagnostics.healthcenter.data.memorycounters=off -Dcom.ibm.diagnostics.healthcenter.data.cpu=off -Dcom.ibm.diagnostics.healthcenter.data.environment=off -Dcom.ibm.diagnostics.healthcenter.data.locking=off -Dcom.ibm.diagnostics.healthcenter.data.memory=off -Dcom.ibm.diagnostics.healthcenter.data.threads=off`;
+    const healthCenterArray = ['bash', '-c', healthCentercommand];
     try {
-      await docker.exec(this.project, ['bash', '-c', command]); 
+      await cwUtils.spawnContainerProcess(this.project, healthCenterArray);
     } catch (error) {
       log.error(error)
     }
 
     const projectID = this.project.projectID;
-
-    let options = {
-      host: "localhost",
-      port: 9090,
-      path: `/api/v1/projects/${projectID}/profiling/${this.metricsFolder}`,
-      method: 'GET',
-    };
-    
-    setTimeout(() => this.getJavaHealthCenterData(options, projectID, this.metricsFolder, 1), duration * 1000);
+    const project = this.project;
+    setTimeout(() => this.getJavaHealthCenterData(project, projectID, this.metricsFolder, 1), duration * 1000);
   }
 
-  async getJavaHealthCenterData(options, projectID, timestamp, counter) {
+  async getJavaHealthCenterData(project, projectID, timestamp, counter) {
```

Comment:
I'm a bit worried about the confusing use of `project` and `this.project` in the same function. We should pick one.
(Also we don't really need to pass `projectID` now we pass the project object, or if it should be the `projectID` for `this.project`.)